Userland/Libraries/LibWeb/Layout/FormattingContext.cpp

```diff
@@ -1640,11 +1640,24 @@ bool FormattingContext::should_treat_height_as_auto(Box const& box, AvailableSpa
     if (computed_height.is_min_content() || computed_height.is_max_content() || computed_height.is_fit_content())
         return true;
 
+    // https://www.w3.org/TR/CSS2/visudet.html#propdef-height
+    // The percentage is calculated with respect to the height of the generated box's containing block.
```

Comment:
Can we find something in https://www.w3.org/TR/css-sizing-3/ to reference instead of CSS2? :)

Reply:
After reading:
https://drafts.csswg.org/css-sizing-3/#cyclic-percentage-contribution
>(...) when calculating the used sizes and positions of the containing block’s contents:
﻿
If the cyclic dependency was introduced due to a [block-size](https://www.w3.org/TR/css-logical-1/#propdef-block-size) or [max-block-size](https://www.w3.org/TR/css-logical-1/#propdef-max-block-size) on the containing block that causes it to depend on the size of its contents, the box’s percentage is not resolved and instead [behaves as auto](https://www.w3.org/TR/css-sizing-3/#behave-as-auto).
﻿
Note: [Grid items](https://www.w3.org/TR/css-grid-2/#grid-item) and [flex items](https://www.w3.org/TR/css-flexbox-1/#flex-item) do allow percentages to resolve in this case.
﻿
Note: These rules specify the previously-undefined behavior of this cyclic case in CSS2§10.2, CSS2§8.3, and CSS2§8.4. Note also, the behavior in **CSS2§10.5** (note from greenclover: the height property in CSS2) is superseded in their respective specifications for layout modes (such as flex layout) not described in CSS2.

https://drafts.csswg.org/css-sizing-3/#funcdef-width-fit-content
> Percentages are resolved against the width/height, as appropriate, of the box’s containing block. If, in a particular axis, the containing block’s size depends on the box’s size, see the **relevant layout module for special rules on how to resolve percentages**.

https://drafts.csswg.org/css-sizing-3/#percentage-sizing
> See § 5.2.1 Intrinsic Contributions of Percentage-Sized Boxes for details on how to resolve percentages when the size of the containing block depends on the size of its content.

https://drafts.csswg.org/css-flexbox-1/#content-based-minimum-size
> For the purpose of calculating an intrinsic size of the box (e.g. the box’s min-content size), a content-based minimum size causes the box’s size in that axis to become indefinite (even if e.g. its width property specifies a definite size). Note this means that percentages calculated against this size will behave as auto.

I've started to wonder if this is the right approach. I'm pretty confused so I'll mark this as a draft for now.